### PR TITLE
tweak api, add linting

### PR DESCRIPTION
**Pull request by mixmix**
This PR:
- :heavy_check_mark: change the DB1 allowPublic signature so that it's more like the DB2 one
- :heavy_check_mark: keeps support for the legacy DB1 signature
- :heavy_check_mark: adds linting
- :heavy_check_mark: does some refactoring to further simplify reading of code

### CODE REVIEW

File: test/core-tests.js [hunk 1]
```diff
@@ -1,39 +1,54 @@
...
+  await p(server.publish)(input)
+    .then(data => t.deepEqual(data.value.content, content, description))
+    .catch(err => t.fail(err, description))
+
+  description = 'legacy: msgs { content, options: { allowPublic: true } }'
+  content = { type: 'profile', name: 'mix' }
+  input = { content, options: { allowPublic: true } }
+  await p(server.publish)(input)
+    .then(data => t.deepEqual(data.value.content, content, description))
+    .catch(err => t.fail(err, description))
```
Comment (mixmix) [inline test/core-tests.js:38]: All these tests are the same, EXCEPT I added this one.

I linearised the tests to make them easier to read

File: README.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ explictly public:
 ```js
 const explicitPublicMsg = {
   content: { type: 'profile' },
-  options: { allowPublic: true }
+  allowPublic: true
 }
```
Comment (mixmix) [inline README.md]: This is what I changed about the API

**Review by Powersource — APPROVED** [review]
The new option format adds more complexity, something which this module already has a lot of, so I would prefer not merging I think. But if you really would like this it's OK.

**Comment by mixmix** [comment]
The alternative is make a breaking change and don't support the legacy way

**Comment by Powersource** [comment]
isn't there also the alternative of not changing the api at all?

**Comment by Powersource** [comment]
2.3.0

**Pull request merged**